zsl/interface/webservice/performers/task.py

```diff
@@ -4,33 +4,37 @@
 """
 from __future__ import unicode_literals
 from builtins import str
-from zsl.application.service_application import service_application as app
-from zsl.router import task_router
 from flask import request
 from flask.helpers import make_response
+
+from zsl import inject, Zsl, Injected
 from zsl.task.job_context import JobContext, WebJobContext
 from zsl.interface.webservice.utils.error_handler import error_handler
 from zsl.interface.webservice.utils.response_headers import append_headers
+from zsl.router.task import TaskRouter
 
 
-@app.route("/task/<path:path>", methods=["POST", "GET", "OPTIONS"])
-@append_headers
-@error_handler
-def perform_web_task(path):
-    app.logger.debug("Performing task %s.", path)
-    if request.method == 'OPTIONS':
-        response = app.make_default_options_response()
-    else:
-        (task, task_callable) = task_router.route(path)
+@inject(app=Zsl)
+def create_web_task(app):
```

Comment:
It's only that all previous code doesn't have them because mostly all the parameters are injected (__init__ methods), so I was just lazy.